src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -558,6 +558,11 @@ public enum ArgumentId
         public static string ReadLengthPrefixedString(BinaryReader reader)
         {
             var length = reader.ReadInt32();
+            if (length < 0)
+            {
+                return null;
```

Comment:
Turn on `nullable` in this file?

Reply:
Don't want to mix changes here. Unwinding this hierarchy is already a complicated rats nest of my own creation. Adding nullable would bog down the change further.

---
In reply to: [319660919](https://github.com/dotnet/roslyn/pull/38411#discussion_r319660919) [](ancestors = 319660919)